proto/api_v3/query_service.proto

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax="proto3";
+
+package jaeger.api_v3;
+
+import "opentelemetry/proto/trace/v1/trace.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+
+option go_package = "api_v3";
+option java_package = "io.jaegertracing.api_v3";
+
+// Request object to get a trace.
+message GetTraceRequest {
+  // Hex encoded 128 bit trace_id
+  string trace_id = 1;
+}
+
+// A single response chunk holds spans from a single trace.
```

Comment:
> To my knowledge, a chunk at the moment always represents a single trace (a trace known at the query time).

I think this is not a matter of how it is currently implemented, but what we want to guarantee.  The intention of the original API was to NOT have this guarantee, i.e. the service is allowed to mix spans from different traces in a single chunk.

Reply:
I think we should guarantee it for these reasons:
* the current API does not mix spans from different traces
* the API that does not mix spans from different traces is easier to consume (e.g. for typical use-cases that we know)

Reply:
I still don't like this guarantee in the API. The streaming & chunking API was primarily introduced for efficiency, but we're taking away server's ability to optimize its response. You could be loading a ton of small traces, e.g. 2 spans each, so this guarantee in the API would force the server to send tiny chunks, which is going to be suboptimal. On the other hand, there is a max chunk size in the server so there is always a possibility that a large trace will be split across several chunks, that largely takes away your "easier to consume" reason #2.

We can always introduce this guarantee later, but removing it would backwards incompatible.

Reply:
In the 5 years of the project history we haven't used the feature of streaming chunks with mixed traces. I don't think any other DT system behaves this way.

>  You could be loading a ton of small traces, e.g. 2 spans each, so this guarantee in the API would force the server to send tiny chunks, which is going to be suboptimal.

This is how Jaeger works right now and we haven't see any complaints/issues/use-case to change it.

Reply:
> We can always introduce this guarantee later, but removing it would backwards incompatible.

It would be a breaking change one way or another. The current consumers do not expect that spans in chunks are mixed (e.g. UI does not).

Reply:
I don't agree with this per my comments above but I have removed the guarantee of not mixing spans in one chunk per Yuri's request.